quantum/quantum_keycodes.h

```diff
@@ -679,15 +679,12 @@ enum quantum_keycodes {
 #define KC_MEH  MEH(KC_NO)
 
 #ifdef UNICODE_ENABLE
-    // For sending unicode codes.
-    // You may not send codes over 7FFF -- this supports most of UTF8.
-    // To have a key that sends out Œ, go UC(0x0152)
-    #define UNICODE(n) (QK_UNICODE | (n))
-    #define UC(n) UNICODE(n)
+  // Allows Unicode input up to 0x7FFF
+  #define UC(c) (QK_UNICODE | (c))
 #endif
-
 #ifdef UNICODEMAP_ENABLE
-    #define X(n) (QK_UNICODE_MAP | (n))
+  // Allows Unicode input up to 0x10FFFF, requires unicode_map
+  #define X(i) (QK_UNICODEMAP | (i))
```

Comment:
I forgot to add this myself, but I'd like to see an "else" for both of these (`UC` and `X`) that define them as empty functions, so they don't error out when compiling when the respective feature is disabled.

Reply:
They could be defined as `KC_NONE` in that case. But what's the use case? It seems to me like you'd want it to error out as fast as possible if you put `UC` or `X` in your keymap but forget to enable the corresponding feature.

Reply:
I'm not sure that failing is the best option here, to be honest. 

Two reasons really come up for this: 
* Community Layouts 
* EEPROM storage of keymaps

Also, testing comes to mind, but that's an edge case. 

The alternative is that if I want to do this locally, I would have to "ifdef" the keycodes ANYWAYS.

Reply:
But for now, I'll just merge this.

Reply:
Could you elaborate on the community layouts bit? I think I know what you mean, but it'd be best if you could clarify just in case.

As far as keymap storage goes, `UC` and `X` themselves aren't part of the keycodes enum (they're just macros), so leaving them undefined doesn't affect other keycodes. What does get left out is `QK_UNICODE` and `QK_UNICODEMAP`, but if I'm not mistaken, those _should_ be left out for the sake of feature flagging.